src/wolf.system/CMakeLists.txt

```diff
@@ -86,121 +102,58 @@ endif()
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
-set (LIBRARY_TYPE "SHARED")
-
-# fetch libexpat
-message("fetching https://github.com/libexpat/libexpat.git")
-FetchContent_Declare(
-  libexpat
-  GIT_REPOSITORY https://github.com/libexpat/libexpat.git
-  GIT_TAG        master
-  SOURCE_SUBDIR  expat
-)
-
-if(WIN32 OR WIN64)
-    # fetch apr
-    message("fetching https://github.com/apache/apr.git")
-    FetchContent_Declare(
-      apr
-      GIT_REPOSITORY https://github.com/apache/apr.git
-      GIT_TAG        trunk
-    )
-endif()
-
-if(WOLF_ENABLE_CURL)
-    message("fetching https://github.com/curl/curl.git")
-    FetchContent_Declare(
-        libcurl
-        GIT_REPOSITORY https://github.com/curl/curl.git
-        GIT_TAG        master
-    )
-endif()
-
-if(WOLF_ENABLE_LUA)
-    message("fetching https://github.com/torch/luajit-rocks.git")
-    FetchContent_Declare(
-        luajit
-        GIT_REPOSITORY https://github.com/torch/luajit-rocks.git
-        GIT_TAG        master
-    )
-endif()
-
-if(WOLF_ENABLE_HTTP1_1_WS)
-    message("fetching https://github.com/madler/zlib.git")
-    FetchContent_Declare(
-        zlib
-        GIT_REPOSITORY https://github.com/madler/zlib.git
-        GIT_TAG        master
-    )
-    message("fetching https://github.com/libuv/libuv.git")
-    FetchContent_Declare(
-        libuv
-        GIT_REPOSITORY https://github.com/libuv/libuv.git
-        GIT_TAG        master
-    )
-endif()
-
-# we need wolf ssl for uSocket and apr socket
-if(WOLF_ENABLE_SSL)
-    message("fetching https://github.com/wolfSSL/wolfssl.git")
-    FetchContent_Declare(
-        wolfssl
-        GIT_REPOSITORY https://github.com/wolfSSL/wolfssl.git
-        GIT_TAG        master
-    )
-endif()
+set(LIBRARY_TYPE "SHARED")
 
 # change some options of CMAKE dependencies
-set(EXPAT_SHARED_LIBS OFF CACHE BOOL "EXPAT_SHARED_LIBS")
-set(EXPAT_BUILD_TESTS OFF CACHE BOOL "EXPAT_BUILD_TESTS")
-set(EXPAT_BUILD_TOOLS OFF CACHE BOOL "EXPAT_BUILD_TOOLS")
 set(CMAKE_DEBUG_POSTFIX "" CACHE STRING "" FORCE)
 set(CMAKE_RELEASE_POSTFIX "" CACHE STRING "" FORCE)
 set(CMAKE_MINSIZEREL_POSTFIX "" CACHE STRING "" FORCE)
 set(CMAKE_RELWITHDEBINFO_POSTFIX "" CACHE STRING "" FORCE)
 set(BUILD_TESTING  OFF CACHE BOOL "BUILD_TESTING")
 
-# change some options of CURL
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "" FORCE)
+endif()
+
+set(EXPAT_SHARED_LIBS OFF CACHE BOOL "EXPAT_SHARED_LIBS")
+set(EXPAT_BUILD_TESTS OFF CACHE BOOL "EXPAT_BUILD_TESTS")
+set(EXPAT_BUILD_TOOLS OFF CACHE BOOL "EXPAT_BUILD_TOOLS")
+
+wolf_find_expat()
```

Comment:
Fetching and building the library `libexapt` currently is only available for windows platforms. Building manually for other platforms is included in the following lines, however, excluding building for the Unix platforms specifies neither here nor at the `external.cmake`.

Reply:
its `FetchContent_Declare` is declared unconditionaly and `FetchContent_MakeAvailable` is also called on `libexpat` on all cases for all platforms:
https://github.com/WolfEngine/WolfEngine/blob/wolf-2/src/wolf.system/CMakeLists.txt#L194-L224 ...